### ISSUE TITLE
gwen console : infinitely recursive error

### ISSUE THREAD

**Issue by Rahul9844** [gwen-interpreter/gwen-web#47]
Below issue is in context to step-definition which has datatable defined.
On pasting/executing gwen dsl without datatable (accidentally), then all subsequent executions of that dsl (with datatable) fails with below error

ERROR - Failed step [at line 1]: And I execute payment flows for "Config.csv": StepDef I execute payment flows for "<PaymentConfigurationFileName>" is infinitely recursive at [line 1]: And I execute payment flows for "Config.csv"
ERROR - [6ms] Failed Step: And I execute payment flows for "Config.csv"

**Comment by bjuric** [comment]
Hi @Rahul9844, can you please provide your StepDef and instructions on how to reproduce this?

**Comment by bjuric** [comment]
Successfully replicated..

Steps to reproduce:
- Load Gwen with a StepDef containing a data table
- Call the step without passing in a data table
  - Gwen reports that data table is expected
- Call the step again passing in a data table
  - Gwen reports infinite recursion error.

```
   __ ___      _____ _ __     _
  / _` \ \ /\ / / _ \ '_ \   { \,"
 | (_| |\ V  V /  __/ | | | {_`/
  \__, | \_/\_/ \___|_| |_|   `
  |___/

gwen-web v-SNAPSHOT

REPL Console

Enter steps to evaluate or type exit to quit..

gwen> And single row without header contains a number in decimal and binary form

INFO - Evaluating Step: And single row without header contains a number in decimal and binary form
gwen.errors.package$DataTableException: Data table expected for StepDef with @DataTable annotation

[non-step]

gwen> :paste

REPL Console (paste mode)

Enter or paste steps and press ctrl-D to evaluate..

       Given single row without header contains a number in decimal and binary form
             | 3 | 11 |

Exiting paste mode, interpreting now..

gwen>        Given single row without header contains a number in decimal and binary form
                   | 3 | 11 |

INFO - Evaluating Step: Given single row without header contains a number in decimal and binary form
ERROR - Failed step [at line 1]: Given single row without header contains a number in decimal and binary form: StepDef single row without header contains a number in decimal and binary form is infinitely recursive at [line 1]: Given single row without header contains a number in decimal and binary form
INFO - Starting chrome browser session
Starting ChromeDriver 2.34.522932 (4140ab217e1ca1bec0c4b4d1b148f3361eb3a03e) on port 15740
Only local connections are allowed.
Jan 23, 2018 5:28:58 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: OSS
ERROR - [8ms] Failed Step: Given single row without header contains a number in decimal and binary form

[Failed]

REPL Console

Enter steps to evaluate or type exit to quit..

```

**Comment by bjuric** [comment]
@Rahul9844, we've implemented a fix for this in the snapshot build below. Please test it in your environment and let us know how it goes.
- [gwen-web-2.15.0-2-g308d08a-SNAPSHOT.zip]( https://oss.sonatype.org/content/repositories/snapshots/org/gweninterpreter/gwen-web/2.15.0-2-g308d08a-SNAPSHOT/gwen-web-2.15.0-2-g308d08a-SNAPSHOT.zip)

**Comment by Rahul9844** [comment]
Error: Could not find or load main class gwen.web.WebInterpreter

**Comment by bjuric** [comment]
@Rahul9844, apologies that one got corrupted. Try this one please:
- [gwen-web-2.15.0-3-g308d08a-SNAPSHOT.zip](https://oss.sonatype.org/content/repositories/snapshots/org/gweninterpreter/gwen-web/2.15.0-3-g308d08a-SNAPSHOT/gwen-web-2.15.0-3-g308d08a-SNAPSHOT.zip)

**Comment by Rahul9844** [comment]
Tested and hence closing as the issue is fixed on gwen-web-2.15.0-3-g308d08a-SNAPSHOT.zip
